core-db/src/main/java/jp/co/soramitsu/core_db/migrations/Migrations.kt

```diff
@@ -2,6 +2,68 @@ package jp.co.soramitsu.core_db.migrations
 
 import androidx.room.migration.Migration
 import androidx.sqlite.db.SupportSQLiteDatabase
+import jp.co.soramitsu.core_db.model.NodeLocal
+import jp.co.soramitsu.core_db.prepopulate.nodes.defaultNodesInsertQuery
+
+class UpdateDefaultNodesList(
+    private val nodesList: List<NodeLocal>,
+    fromVersion: Int,
+) : Migration(fromVersion, fromVersion + 1) {
+
+    init {
+        require(nodesList.all { it.isActive.not() },) {
+            "Nodes should not be active by default"
+        }
+    }
+
+    override fun migrate(database: SupportSQLiteDatabase) {
+        database.beginTransaction()
+
+        val activeNodeLinkCursor = database.query("SELECT networkType FROM nodes WHERE isActive = 1")
+
+        val activeNodeNetworkType = if (activeNodeLinkCursor.moveToNext()) {
+            val networkType = activeNodeLinkCursor.getInt(activeNodeLinkCursor.getColumnIndex("networkType"))
+
+            networkType
+        } else null
+
+        activeNodeLinkCursor.close()
+
+        val modifiedNodesList = if (activeNodeNetworkType != null) {
+            val mutableNodesList = nodesList.toMutableList()
+
+            val firstRelevantDefaultNode = nodesList.first { it.networkType == activeNodeNetworkType && it.isDefault }
+            val indexOfRelevantNode = nodesList.indexOf(firstRelevantDefaultNode)
+
+            mutableNodesList[indexOfRelevantNode] = firstRelevantDefaultNode.copy(isActive = true)
+
+            mutableNodesList
+        } else {
+            nodesList
+        }
+
+        database.execSQL("DELETE FROM nodes WHERE isDefault = 1")
+        database.execSQL(defaultNodesInsertQuery(modifiedNodesList))
+
+        database.setTransactionSuccessful()
+        database.endTransaction()
+    }
+}
+
+@Suppress("ClassName")
+class MoveActiveNodeTrackingToDb_18_19(private val migrator: PrefsToDbActiveNodeMigrator) : Migration(18, 19) {
```

Comment:
And here we have migration from 18 to 19, but in another file we have version 20. Can you please help me with understanding why it is so? 🙃

Reply:
In this PR i, in general, did two things:
1. Move active node from prefs to DB
2. Change default node set

I decided to do them in separate migration for readability and easier understanding of migrations. That's why there are 18 -> 19 and 19 -> 20 migrations